internal/compiler/fileloader.go

```diff
@@ -221,9 +224,31 @@ func (t *parseTask) start(loader *fileLoader) {
 	})
 }
 
+func (p *fileLoader) CacheSourceFileMetaData(path tspath.Path) {
+	p.sourceFileMetaDatasMutex.Lock()
+	defer p.sourceFileMetaDatasMutex.Unlock()
```

Comment:
This lock is held for the entire call, so I don't think we're gaining any parallelism here, no?

The "fix" would be to only hold the lock for the check at the top, unlock it, then relock it at the bottom (moving the make call down too).

Reply:
Which, also indicates that this should be a RWLock.